packages/ui-drilldown/src/Drilldown/__tests__/Drilldown.test.tsx

```diff
@@ -23,105 +23,532 @@
+      const onToggle = stub()
+      await mount(
+        <Drilldown
+          rootPageId="page0"
+          trigger={<button>Options</button>}
+          onToggle={onToggle}
+          defaultShow
+        >
+          <Drilldown.Page id="page0">
+            <Drilldown.Option id="option0">Option 0</Drilldown.Option>
+          </Drilldown.Page>
+        </Drilldown>
+      )
       const drilldown = await DrilldownLocator.find()
+      await drilldown.keyUp('esc')
 
-      expect(drilldown).to.exist()
+      expect(onToggle).to.have.been.calledOnce()
+      expect(onToggle).to.have.been.calledWithMatch(match.instanceOf(Event), {
+        shown: false,
+        drilldown: match.object,
+        pageHistory: ['page0'],
+        goToPage: match.func,
+        goToPreviousPage: match.func
+      })
     })
+    it('should call onDismiss when Drilldown is closed', async () => {
+      const onDismiss = stub()
+      await mount(
+        <Drilldown
+          rootPageId="page0"
+          trigger={<button>Options</button>}
+          onDismiss={onDismiss}
+          defaultShow
+        >
+          <Drilldown.Page id="page0">
+            <Drilldown.Option id="option0">Option 0</Drilldown.Option>
+          </Drilldown.Page>
+        </Drilldown>
+      )
+      const drilldown = await DrilldownLocator.find()
 
-    describe('for a11y', async () => {
-      it('should be accessible', async () => {
-        await mount(
-          <Drilldown rootPageId="page0">
-            <Drilldown.Page
-              id="page0"
-              renderTitle="Page Title"
-              renderActionLabel="Action Label"
-            >
-              <Drilldown.Option id="item01">Item1</Drilldown.Option>
-              <Drilldown.Option id="item02" subPageId="page1">
-                Item2
-              </Drilldown.Option>
-              <Drilldown.Option id="item03" description="This is a description">
-                Item3
-              </Drilldown.Option>
-              <Drilldown.Option id="item04" renderLabelInfo="After">
-                Item4
-              </Drilldown.Option>
-              <Drilldown.Option id="item05" disabled>
-                Item5
-              </Drilldown.Option>
-              <Drilldown.Option id="item06" href="/">
-                Item6
-              </Drilldown.Option>
-              <Drilldown.Option
-                id="item07"
-                renderBeforeLabel={<IconCheckSolid />}
-              >
-                Item7
-              </Drilldown.Option>
-              <Drilldown.Option
-                id="item08"
-                renderAfterLabel={<IconCheckSolid />}
-              >
-                Item8
+      await drilldown.keyUp(27)
+
+      expect(onDismiss).to.have.been.calledOnce()
+      expect(onDismiss).to.have.been.calledWithMatch(
+        match.instanceOf(Event),
+        false
+      )
+    })
+  })
+
+  it('should give back the correct reference when "ref" is set', async () => {
```

Comment:
Please remove this too, I've written ref tests, and also I made a bugfix here so this is not valid anymore (the ref shouldn't be the trigger but the popover root element)